File: source_funcs/go_transcoder/MAKING_ELEMENTS_OF_TWO_ARRAYS_SAME_WITH_MINIMUM_INCREMENTDECREMENT/MAKING_ELEMENTS_OF_TWO_ARRAYS_SAME_WITH_MINIMUM_INCREMENTDECREMENT.go
package main

import (
	"math"
	"sort"
)

func f_gold(a []int, b []int, n int) int {
	sort.Ints(a)
	sort.Ints(b)
	var result int = 0
	for i := int(0); i < n; i++ {
		result = result + int(math.Abs(float64(a[i]-b[i])))
	}
	return result
}

File: rust_funcs/cpp_transcoder/MAKING_ELEMENTS_OF_TWO_ARRAYS_SAME_WITH_MINIMUM_INCREMENTDECREMENT/MAKING_ELEMENTS_OF_TWO_ARRAYS_SAME_WITH_MINIMUM_INCREMENTDECREMENT.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::cmp::Ordering;

fn f_gold(a: &mut [f32], b: &mut [f32], n: i32) -> i32 { 
    a.sort_by(|a, b| a.partial_cmp(b).unwrap());
    b.sort_by(|a, b| a.partial_cmp(b).unwrap());
    let mut result = 0;
    for i in 0..n as usize {
        result += (a[i] - b[i]).abs() as i32;
    }
    result
}
